internal/db/queryBuilder.go

```diff
@@ -185,3 +208,17 @@ func (qb *queryBuilder) build() (string, []any) {
 
 	return query, args
 }
+
+func (qb *queryBuilder) build(asCte bool) (string, []any) {
```

Comment:
Didn't think a separate method was necessary so I added an arg to return the query as a CTE instead of a select statement